### PR TITLE
OAuth2PermissionGrants POST doc

### PR DESCRIPTION
**Pull request by bpcastilho**
Adding Post for OAuth2PermissionGrants

### CODE REVIEW

File: api-reference/beta/api/oauth2permissiongrant-post.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+---
+title: "Create oAuth2PermissionGrant"
+description: "Use this API to create a new oAuth2PermissionGrant."
+localization_priority: Normal
+author: ""
```
Comment (dkershaw10) [inline api-reference/beta/api/oauth2permissiongrant-post.md]: Make this davidmu1

File: api-reference/beta/api/oauth2permissiongrant-post.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ms.prod: "microsoft-identity-platform"
+doc_type: "apiPageType"
+---
+
+# Create oAuth2PermissionGrant
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Use this API to create a new oAuth2PermissionGrant.
```
Comment (dkershaw10) [inline api-reference/beta/api/oauth2permissiongrant-post.md]: Should be a link to the resource topic.

File: api-reference/beta/api/oauth2permissiongrant-post.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+| Delegated (work or school account) | DelegatedPermissionGrant.ReadWrite.All, Directory.AccessAsUser.All |
+| Delegated (personal Microsoft account) | Not supported. |
+| Application                            | DelegatedPermissionGrant.ReadWrite.All |
+
+## HTTP request
+
+<!-- { "blockType": "ignored" } -->
+
+```http
+POST /oauth2PermissionGrants/{id}
```
Comment (dkershaw10) [inline api-reference/beta/api/oauth2permissiongrant-post.md]: Remove /{id}

File: api-reference/beta/api/oauth2permissiongrant-post.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+```
+
+## Request headers
+
+| Name          | Description   |
+|:--------------|:--------------|
+| Authorization | Bearer {token} |
```
Comment (dkershaw10) [inline api-reference/beta/api/oauth2permissiongrant-post.md]: Add Content-Type: application/json

File: api-reference/beta/api/oauth2permissiongrant-post.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+## Request body
+
+In the request body, supply a JSON representation of [oAuth2PermissionGrant](../resources/oauth2permissiongrant.md) object.
+
+## Response
+
+If successful, this method returns `201, Created` response code and a new [oAuth2PermissionGrant](../resources/oauth2permissiongrant.md) object in the response body.
```
Comment (dkershaw10) [inline api-reference/beta/api/oauth2permissiongrant-post.md]: 201, Created -> 201 Created
[bug in the generation tooling]

Comment (dkershaw10) [inline api-reference/beta/api/oauth2permissiongrant-post.md]: 201, Created -> 201 Created
[bug in the generation tooling]

File: api-reference/beta/api/oauth2permissiongrant-post.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+}-->
+
+```http
+POST https://graph.microsoft.com/beta/oauth2PermissionGrants
+Content-type: application/json
+
+{
+  "clientId": "clientId-value",
+  "consentType": "consentType-value",
+  "expiryTime": "datetime-value",
```
Comment (dkershaw10) [inline api-reference/beta/api/oauth2permissiongrant-post.md]: startTime and expiryTime might be in beta, but they are being remove in v1.0, so hopefully neither are required.

File: api-reference/beta/api/oauth2permissiongrant-post.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
+---
+title: "Create oAuth2PermissionGrant"
```
Comment (dkershaw10) [inline api-reference/beta/api/oauth2permissiongrant-post.md]: I'm sure the entity type is called oauth2PermissionGrant.
Not sure where the capital A came from

Comment (bpcastilho) [reply api-reference/beta/api/oauth2permissiongrant-post.md]: There are tons of other places with the capital A, so I will changed them all.

Comment (bpcastilho) [reply api-reference/beta/api/oauth2permissiongrant-post.md]: It looks like we are using the capital A for the rest of the docs, soo I'll keep it.

File: api-reference/beta/api/oauth2permissiongrant-post.md (hunk 6 above)
Comment (bpcastilho) [reply api-reference/beta/api/oauth2permissiongrant-post.md]: Tested removing expiryTime and that's required in Beta.

**Pull request closed without merging**